auth0/src/main/java/com/auth0/android/provider/OAuthManager.java

```diff
@@ -0,0 +1,295 @@
+    private int requestCode;
+    private PKCE pkce;
+
+    public OAuthManager(Auth0 account, AuthCallback callback, Map<String, String> parameters) {
+        this.account = account;
+        this.callback = callback;
+        this.parameters = new HashMap<>(parameters);
+    }
+
+    public void useFullScreen(boolean useFullScreen) {
+        this.useFullScreen = useFullScreen;
+    }
+
+    public void useBrowser(boolean useBrowser) {
+        this.useBrowser = useBrowser;
+    }
+
+    public void setLoggingEnabled(boolean enabled) {
+        this.loggingEnabled = enabled;
+    }
+
+    @VisibleForTesting
+    void setPKCE(PKCE pkce) {
+        this.pkce = pkce;
+    }
+
+    public void startAuthorization(Activity activity, String redirectUri, int requestCode) {
+        addPKCEParameters(parameters, redirectUri);
+        addClientParameters(parameters, redirectUri);
+        addValidationParameters(parameters);
+        Uri uri = buildAuthorizeUri();
+
+        this.requestCode = requestCode;
+        final Intent intent;
+        if (this.useBrowser) {
+            Log.d(TAG, "About to start the authorization using the Browser");
+            intent = new Intent(Intent.ACTION_VIEW, uri);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+            activity.startActivity(intent);
+        } else {
+            Log.d(TAG, "About to start the authorization using the WebView");
+            intent = new Intent(activity, WebAuthActivity.class);
+            intent.setData(uri);
+            intent.putExtra(WebAuthActivity.CONNECTION_NAME_EXTRA, parameters.get(KEY_CONNECTION));
+            intent.putExtra(WebAuthActivity.FULLSCREEN_EXTRA, useFullScreen);
+            activity.startActivityForResult(intent, requestCode);
+        }
+    }
+
+    public boolean resumeAuthorization(AuthorizeResult data) {
+        if (!data.isValid(requestCode)) {
+            Log.w(TAG, "The Authorize Result is invalid.");
+            return false;
+        }
+
+        final Map<String, String> values = CallbackHelper.getValuesFromUri(data.getIntent().getData());
+        if (values.isEmpty()) {
+            Log.w(TAG, "The response didn't contain any of these values: code, state, id_token, access_token, token_type, refresh_token");
+            return false;
+        }
+        logDebug("The parsed CallbackURI contains the following values: " + values);
+
+        try {
+            checkErrorValue(values.get(KEY_ERROR));
+            checkStateValue(values.get(KEY_STATE));
+            checkNonceValue(values.get(KEY_NONCE), values.get(KEY_ID_TOKEN));
+        } catch (AuthenticationException e) {
+            callback.onFailure(e);
+            return true;
+        }
+
+        Log.d(TAG, "Authenticated using web flow");
+        final Credentials urlCredentials = new Credentials(values.get(KEY_ID_TOKEN), values.get(KEY_ACCESS_TOKEN), values.get(KEY_TOKEN_TYPE), values.get(KEY_REFRESH_TOKEN));
+        if (!shouldUsePKCE()) {
```

Comment:
There is little gain here to early exit since is just a method invocation so we can use a simple if-else